astacus/coordinator/plugins/clickhouse/steps.py

```diff
@@ -190,14 +190,14 @@ async def revoke_write_on_table(self, table: Table, user_name: bytes) -> None:
         revoke_statement = (
             f"REVOKE INSERT, ALTER UPDATE, ALTER DELETE ON {table.escaped_sql_identifier} FROM {escaped_user_name}"
         )
-        await asyncio.gather(*(client.execute(revoke_statement.encode()) for client in self.clients))
+        await self.clients[0].execute(revoke_statement.encode())
 
     async def grant_write_on_table(self, table: Table, user_name: bytes) -> None:
         escaped_user_name = escape_sql_identifier(user_name)
         grant_statement = (
             f"GRANT INSERT, ALTER UPDATE, ALTER DELETE ON {table.escaped_sql_identifier} TO {escaped_user_name}"
         )
-        await asyncio.gather(*(client.execute(grant_statement.encode()) for client in self.clients))
+        await self.clients[0].execute(grant_statement.encode())
```

Comment:
This change looks very controversial to me
1. Yes, in principle it is not necessary to change priileges multiple times, but we do not control how fast it will reach other servers, do we? Theoretically it might be delayed for unknown time which spoils all guarantees.
2. I really want to continue [this](https://aiven-io.slack.com/archives/C052RDBD1TP/p1730911269847339) discussion. So far I think it is better to just delete this step completely...

Reply:
I mean the cluster is in a single region, and zookeeper is strictly serializable, so the upper bound on the latency for the change to go through in ZK is <5ms.

Actually I don't think the deduplication log is relevant for the restore. The log is saved in ZooKeeper, and the restore process doesn't start from a ZK snapshot but from an empty state. So the block-level information is lost.

Reply:
Will test if in practice the argument I just provided stands, but the main branch is broken at the moment and I'd rather we fix it. If the step is irrelevant, then of course we'll drop it.

Reply:
Zookeeper only supports write operation serialization, but reads could propagate slowly.
They first go to the local zookeeper, and then to the clickhouse to be processed, as I understand.
But looks like it does not efficiently make things worse, you are right.

2. the solution is not easy, so, lets start with this one